### PR TITLE
Prepare Azure.Core 1.14.0 release

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.13.0-beta.1 (Unreleased)
+## 1.14.0 (2021-04-07)
```
Comment (jsquire) [inline sdk/core/Azure.Core/CHANGELOG.md:3]: You're skipping 1.13.0 and going right to 1.14 - was that intentional?

**Review by jsquire — APPROVED** [review]

File: sdk/core/Azure.Core/CHANGELOG.md (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core/CHANGELOG.md:3]:
```suggestion
## 1.13.0 (2021-04-07)
```

File: sdk/core/Azure.Core/src/Azure.Core.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <Description>This is the implementation of the Azure Client Pipeline</Description>
     <AssemblyTitle>Microsoft Azure Client Pipeline</AssemblyTitle>
-    <Version>1.13.0-beta.1</Version>
+    <Version>1.14.0</Version>
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Azure.Core.csproj:5]:
```suggestion
    <Version>1.13.0</Version>
```

**Pull request merged**